lib/internal/modules/package_json_reader.js

```diff
@@ -1,60 +1,37 @@
 'use strict';
 
 const {
-  JSONParse,
-  ObjectPrototypeHasOwnProperty,
-  SafeMap,
   StringPrototypeEndsWith,
   StringPrototypeIndexOf,
   StringPrototypeLastIndexOf,
   StringPrototypeSlice,
 } = primordials;
-const {
-  ERR_INVALID_PACKAGE_CONFIG,
-} = require('internal/errors').codes;
-const { internalModuleReadJSON } = internalBinding('fs');
+const modulesBinding = internalBinding('modules');
 const { resolve, sep, toNamespacedPath } = require('path');
 const permission = require('internal/process/permission');
-const { kEmptyObject, setOwnProperty } = require('internal/util');
-
-const { fileURLToPath, pathToFileURL } = require('internal/url');
-
-const cache = new SafeMap();
+const { kEmptyObject } = require('internal/util');
 
 let manifest;
-
-/**
- * @typedef {{
- *   exists: boolean,
- *   pjsonPath: string,
- *   exports?: string | string[] | Record<string, unknown>,
- *   imports?: string | string[] | Record<string, unknown>,
- *   name?: string,
- *   main?: string,
- *   type: 'commonjs' | 'module' | 'none',
- * }} PackageConfig
- */
-
 /**
  * @param {string} jsonPath
  * @param {{
- *   base?: string,
- *   specifier: string,
- *   isESM: boolean,
+ *   base?: URL | string,
+ *   specifier?: URL | string,
+ *   isESM?: boolean,
  * }} options
- * @returns {PackageConfig}
+ * @returns {import('typings/internalBinding/modules').PackageConfig}
  */
 function read(jsonPath, { base, specifier, isESM } = kEmptyObject) {
-  if (cache.has(jsonPath)) {
-    return cache.get(jsonPath);
-  }
-
-  const string = internalModuleReadJSON(
+  // TODO(@anonrig): Understand why specifier and base is sometimes URL and sometimes string.
+  const parsed = modulesBinding.readPackageJSON(
     toNamespacedPath(jsonPath),
+    isESM,
+    base == null ? undefined : `${base}`,
```

Comment:
```suggestion
    base == null ? undefined : base,
```

Reply:
base can be URL as well

Reply:
Ahhh, wdyt about renaming base to baseOrURL then?

Reply:
I'm more interested about calling this function with only string parameter rather than string or URL distinction